### PR TITLE
Sarkars/unify rewrite pass

### PR DESCRIPTION
**Pull request by sayantan-nervana**
Merge `ngraph_rewrite_pass` from vars-and-optimizers path to normal path

### CODE REVIEW

File: src/ngraph_rewrite_pass.cc [hunk 1]
```diff
@@ -209,6 +214,15 @@ class NGraphEncapsulationPass : public NGraphRewritePass {
       return Status::OK();
     }
 
+#if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
```
Comment (sayantan-nervana) [inline src/ngraph_rewrite_pass.cc]: Technically we dont need the ifdef perhaps? `ReplaceModifiers` should not have any effect in grappler or normal paths

File: src/ngraph_rewrite_pass.cc [hunk 2]
```diff
@@ -248,6 +262,15 @@ class NGraphEncapsulationPass : public NGraphRewritePass {
                  "Graph with Variables Rewritten for Tracking");
     }
 
+#if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
```
Comment (sayantan-nervana) [inline src/ngraph_rewrite_pass.cc]: Similarly, this could run in the normal/grappler path without any ill-effects right? so the ifdef could be removed

File: src/ngraph_rewrite_pass.cc (hunk 1 above)
Comment (kanvi-nervana) [reply src/ngraph_rewrite_pass.cc]: Yes, perhaps but I would suggest to keep the ifdef for clarity.

File: src/ngraph_rewrite_pass.cc (hunk 2 above)
Comment (kanvi-nervana) [reply src/ngraph_rewrite_pass.cc]: Sam as above.

File: src/ngraph_encapsulate_op.cc [hunk 3]
```diff
@@ -44,8 +44,8 @@
 #endif
 
 #if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
-#include "enable_variable_ops/ngraph_catalog.h"
-#include "enable_variable_ops/ngraph_var.h"
+#include "ngraph_catalog.h"
+#include "ngraph_var.h"
```
Comment (shresthamalik) [inline src/ngraph_encapsulate_op.cc]: Why were these moved out?

File: src/ngraph_rewrite_pass.cc [hunk 4]
```diff
@@ -29,6 +29,11 @@
 #include "ngraph_utils.h"
 #include "tf_graph_writer.h"
 
+#if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
+#include "ngraph_enter_in_catalog.h"
+#include "ngraph_replace_variable_modifiers.h"
```
Comment (shresthamalik) [inline src/ngraph_rewrite_pass.cc]: They can be included from outside the enable_variable_ops/ ?

Comment (sayantan-nervana) [reply src/ngraph_rewrite_pass.cc]: Generally i was trying to move out everything I could from the `src/enable_variable_ops` folder to `src` in the general effort to merge normal + enable_variable_ops

Are you saying we merge files in normal + enable_variable_ops that already existed in normal, but for new files (like `ngraph_enter_in_catalog.h`) we keep them in `src/enable_variable_ops`. That is a good idea too, in case we want to sort of logically partition `src` so that `src` doesn't become a huge file dump?
